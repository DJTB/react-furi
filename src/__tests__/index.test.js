// import 'jest-styled-components';
import React from 'react';
// import { render } from 'enzyme';

import ReactFuri from '../index';

const render = () => 'FIXME';

describe('<ReactFuri />', () => {
  it('no furi provided: render entire reading over kanji block', () => {
    const renderedComponent = render(<ReactFuri word="漢字" reading="かんじ" />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('furi provided: render readings over related kanji blocks', () => {
    const renderedComponent = render(<ReactFuri word="漢字" reading="かんじ" furi="0:かん;1:じ" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
