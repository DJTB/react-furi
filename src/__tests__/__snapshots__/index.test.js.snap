// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ruby /> furi provided: render readings over related kanji blocks 1`] = `
.c0 {
  padding: 0.225rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c2 {
  font-size: 0.95em;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
  padding-bottom: 0.2em;
  opacity: 0.9;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  font-size: 2.3em;
}

@media (min-width:600px) {
  .c0 {
    padding: 0.33325rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    lang="ja"
  >
    <div
      class="c2"
    >
      かん
    </div>
    <div
      class="c3"
    >
      漢
    </div>
  </div>
  <div
    class="c1"
    lang="ja"
  >
    <div
      class="c2"
    >
      じ
    </div>
    <div
      class="c3"
    >
      字
    </div>
  </div>
</div>
`;

exports[`<Ruby /> no furi provided: render entire reading over kanji block 1`] = `
.c0 {
  padding: 0.225rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c2 {
  font-size: 0.95em;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
  padding-bottom: 0.2em;
  opacity: 0.9;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3 {
  font-size: 2.3em;
}

@media (min-width:600px) {
  .c0 {
    padding: 0.33325rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    lang="ja"
  >
    <div
      class="c2"
    >
      かんじ
    </div>
    <div
      class="c3"
    >
      漢字
    </div>
  </div>
</div>
`;
